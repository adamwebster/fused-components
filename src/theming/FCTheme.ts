import React from 'react';

export interface Theme {
    theme?: string,
    children?: any,
}

export const FCTheme = React.createContext<Theme | null>(null);

export const FCThemeProvider = FCTheme.Provider;

export const FCThemeConsumer = FCTheme.Consumer;