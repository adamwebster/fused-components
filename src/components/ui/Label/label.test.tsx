import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-styled-components';
import { Label } from './index';
import { FCThemeProvider } from '../../../theming/FCTheme';

afterEach(cleanup);

describe('Label Tests', () => {
  test('Renders the Label component', () => {
    const { getByText } = render(<Label htmlFor="input">I am a label</Label>);
    expect(getByText('I am a label')).toBeInTheDocument();
  });

  test('Renders the for attribute', () => {
    const { getByText } = render(<Label htmlFor="input">I am a label</Label>);
    expect(getByText('I am a label').getAttribute('for')).toBe('input');
  });
  test('Renders the Label component | Dark mode', () => {
    const { container } = render(
      <FCThemeProvider value={{ theme: 'dark' }}>
        <Label>I am a label</Label>
      </FCThemeProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
