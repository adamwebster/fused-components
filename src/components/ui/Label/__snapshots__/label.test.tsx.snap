// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Tests Renders the Label component | Dark mode 1`] = `
.c0 {
  display: inline-block;
  color: #DDDDDD;
  margin-bottom: 5px;
  font-weight: bold;
}

<div>
  <label
    class="c0"
  >
    I am a label
  </label>
</div>
`;
